opal/mca/pmix/base/pmix_base_fns.c

```diff
@@ -193,20 +193,22 @@ static void translate_params(void)
     home = opal_home_directory();
     if (NULL != home) {
         file = opal_os_path(false, home, ".openmpi", "mca-params.conf", NULL);
-        OBJ_CONSTRUCT(&params, opal_list_t);
-        mca_base_parse_paramfile(file, &params);
-        free(file);
-        OPAL_LIST_FOREACH (fv, &params, mca_base_var_file_value_t) {
-            pmix_overlap = check_pmix_overlap(&e2[len], evar);
-            if (!pmix_overlap && check_pmix_param(fv->mbvfv_var)) {
-                opal_asprintf(&tmp, "PMIX_MCA_%s", fv->mbvfv_var);
-                // set it, but don't overwrite if they already
-                // have a value in our environment
-                setenv(tmp, fv->mbvfv_value, false);
-                free(tmp);
+        if (NULL != file) {
+            OBJ_CONSTRUCT(&params, opal_list_t);
+            mca_base_parse_paramfile(file, &params);
+            free(file);
+            OPAL_LIST_FOREACH (fv, &params, mca_base_var_file_value_t) {
+               pmix_overlap = (NULL == evar) ? 0 : check_pmix_overlap(fv->mbvfv_var, evar);
```

Comment:
Please see my comment on your mpi4py PR - this is an incorrect fix. The `evar` variable is not set - you are using a dangling value from the prior code block.

Reply:
hu? the triad has check for evar being null

Reply:
okay i see what you're saying.  we just shouldn't use evar in this section.